### ISSUE TITLE
Support server side rendering

### ISSUE THREAD

**Issue by hawkins** [bharley/react-window-dimensions#1]
First off, thanks for this! It's a nice and handy HOC! :1st_place_medal: 

---

My only trouble with this component is that it doesn't support server side rendering, because of its strong ties to the `window` object. I think something as simple as `window = window || { innerWidth: 800, innerHeight: 600};` might be just enough to support this if only for an initial render, but that involves a default window size that may not be useful for first-renders, causing components to re-render once theyre on the client with the corrected size.

I'm not entirely sure what all goes into supporting server side rendering, but I've forked the project to see if I can manage it quickly.

**Comment by hawkins** [comment]
Of course, the event listeners are the actual issue here, and those seem to stop anyone from server-rendering components like this. I'm investigating code splitting and creating a code split around this component now.

**Comment by insipx** [comment]
what about just adding a `if (typeof window !== 'undefined') wherever you use window?

**Comment by hawkins** [comment]
That would work for simpler applications, but since this library depends on window event listeners, simply blocking out that logic from server renderers causes the library to not be responsive in the front end, as there are no event listeners to listen to the window resizing.

So, I believe that means we would need a way to get this logic injected after the fact, i.e. lazy loading or dynamic imports.

**Comment by hawkins** [comment]
For what it's worth, code splitting around it, i.e. dynamic import / lazy load, was the key to using this in a server-side rendered setting. My component using this library would load and was able to appear fully functional but would not actually handle resize until the dynamic import was processed. YMMV

**Comment by giovannipds** [comment]
That's a big issue, I'll not use the component just because of it, unfortunately.

**Comment by giovannipds** [comment]
It'll be good to inform that on readme until not supported.
